.github/workflows/lint.yml

```diff
@@ -0,0 +1,25 @@
+name: lint
+on:
+  pull_request:
+    branches: ["**"]
+
+env:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+  DOTNET_NOLOGO: true
+
+jobs:
+  lint_csharp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "6.0.x"
```

Comment:
what version of dotnet should we actually use?
https://github.com/hashicorp/vault-client-dotnet/blob/main/.github/workflows/release.yml#L12

Reply:
the `dotnet format` tool was only available as of `5.0.0` and up.

wrt to releases and testing, we only use .Net for the build and test tooling so it doesn't really matter as long as it supports what we need. Likely we should change this all to latest `.Net 6`. 

When we build the library the version of .net is `.net standard`, which means that it's compatible with a good chunk of frameworks.

Reply:
I'll change them in release and testing to make sure they're more consistent.